presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -475,7 +484,23 @@ public CounterStat getHighMemorySplitSource()
 
                 results.add(new HivePartitionMetadata(hivePartition, Optional.of(partition), partitionSchemaDifference.build(), encryptionInformation));
             }
-
+            StringBuilder warningMessage = new StringBuilder(format("Table '%s' has '%s' out of '%s' partitions unreadable: ", tableName, unreadablePartitionsSkipped, partitionBatch.size()));
+            for (Entry<String, Set<String>> entry : partitionsNotReadable.entrySet()) {
+                if (entry.getValue().size() > 2) {
+                    Iterator it = entry.getValue().iterator();
+                    warningMessage.append("\n" + entry.getKey().toString() + ": " + it.next() + ", " + it.next() + " ...");
+                }
+                else {
+                    warningMessage.append("\n" + entry.getKey().toString() + ": " + entry.getValue().toString());
+                }
+            }
+            if (partitionsNotReadable.entrySet().size() > 2) {
+                if (warningMessage.lastIndexOf("\n") >= 0) {
+                    warningMessage.delete(warningMessage.lastIndexOf("\n"), warningMessage.length());
+                    warningMessage.append("\n...");
+                }
+            }
```

Comment:
prior to this if statement, max of 3 reasons in the warningMessage. If 3, delete the last one and append "..."